cpp/libs/ossrf_gstreamer_api/lib/src/sender/st2110_30_sender_plugin.cpp

```diff
@@ -0,0 +1,137 @@
+        BST_ENFORCE(audioresample != nullptr, "Failed creating GStreamer element audioresample");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), audioresample), "Failed adding audioresample to the pipeline");
+
+        // Add pipeline capsfilter
+        auto* capsfilter = gst_element_factory_make("capsfilter", NULL);
+        BST_ENFORCE(capsfilter != nullptr, "Failed creating capsfilter");
+
+        // Create caps for capsfilter
+        auto* caps = gst_caps_new_simple("audio/x-raw", "channels", G_TYPE_INT, f_.number_of_channels, "rate",
+                                         G_TYPE_INT, f_.sampling_rate, NULL);
+        BST_ENFORCE(caps != nullptr, "Failed creating GStreamer audio caps");
+        g_object_set(G_OBJECT(capsfilter), "caps", caps, NULL);
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), capsfilter), "Failed adding capsfilter to the pipeline");
+        gst_caps_unref(caps);
+
+        // Add pipeline queue1
+        auto* queue3 = gst_element_factory_make("queue", NULL);
+        BST_ENFORCE(queue3 != nullptr, "Failed creating GStreamer element queue");
+        g_object_set(G_OBJECT(queue1), "max-size-time", queue_max_size_time, "max-size-buffers", queue_max_size_buffers,
+                     "max-size-bytes", queue_max_size_bytes, NULL);
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), queue3), "Failed adding queue to the pipeline");
+
+        // Add pipeline rtpL24pay (audio payload)
+        auto* rtpL24pay = gst_element_factory_make("rtpL24pay", NULL);
+        BST_ENFORCE(rtpL24pay != nullptr, "Failed creating GStreamer element rtpL24pay");
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), rtpL24pay), "Failed adding rtpL24pay to the pipeline");
+
+        // Add pipeline udpsink
+        auto* udpsink = gst_element_factory_make("udpsink", NULL);
+        BST_ENFORCE(udpsink != nullptr, "Failed creating GStreamer element udpsink");
+        // Set properties
+        g_object_set(G_OBJECT(udpsink), "host", s_.primary.destination_ip_address.c_str(), NULL);
+        g_object_set(G_OBJECT(udpsink), "port", s_.primary.destination_port, NULL);
+        g_object_set(G_OBJECT(udpsink), "auto-multicast", TRUE, NULL);
+        g_object_set(G_OBJECT(udpsink), "multicast-iface", s_.primary.interface_name.c_str(), NULL);
+        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), udpsink), "Failed adding udpsink to the pipeline");
+
+        // Link elements
+        BST_ENFORCE(gst_element_link_many(source, queue1, audioconvert, queue2, audioresample, capsfilter, queue3,
+                                          rtpL24pay, udpsink, NULL),
+                    "Failed linking GStreamer audio pipeline");
+
+        // Setup runner
+        pipeline_.run_loop();
+
+        return {};
+    }
+
+    void stop() noexcept override
+    {
+        pipeline_.stop();
+        pipeline_ = {};
+    }
+};
+
+// TODO this function will need to receive an SDP and use the information in it to build the GST pipeline
```

Comment:
This does not apply to the sender just for the receiver, please remove this TODO